src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java

```diff
@@ -53,6 +58,29 @@ protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
 		String translation = ClientUtils.localize(SCContent.BLOCK_POCKET_MANAGER.get().getTranslationKey());
 
 		font.drawString(translation, xSize / 2 - font.getStringWidth(translation) / 2, 6, 4210752);
+
+		if (!te.enabled)
+		{
+			font.drawString(ClientUtils.localize("gui.securitycraft:blockPocketManager.youNeed"), 7, 83, 4210752);
+
+			font.drawString((size - 2) * (size - 2) * 6 + "", 42, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(new ItemStack(SCContent.BLOCK_POCKET_WALL.get()), 25, 96);

```

Comment:
Please don't create a new ItemStack every tick. Instead, create private static final fields at the top of this class that contain all the ItemStacks you use in this method, and use those fields instead.

Reply:
There are still some ItemStacks further down.